### DISCUSSION TITLE
Add warming support

### DISCUSSION THREAD

**Discussion by Jamesking56** [stefanzweifel/sidecar-browsershot#16] [Ideas]
In my use case, I'd like to add warming to the Browsershot Lambda so that requests are quicker but there doesn't seem to be an easy way to configure it:

https://hammerstone.dev/sidecar/docs/main/functions/warming

**Comment by stefanzweifel** [comment]
+1 on that!
I've wondered when the first message comes in about warming support. 😅
(I've also just now realised, that by default a Sidecar function doesn't support warming as the default number of instances is 0. No wonder my setup on [3.screeenly.com](https://3.screeenly.com/playground) didn't work. 🙃)

I've been thinking for a while how we could accomplish this. The problem is, that the `BrowsershotFunction`-class is not resolved out of the IOC-container, but just "called" inside `BrowsershotLambda` [here](https://github.com/stefanzweifel/sidecar-browsershot/blob/main/src/BrowsershotLambda.php#L24).

If the class would be injected, you could change the [binding](https://laravel.com/docs/master/container#binding) and pass a `MyBrowsershotFunction` that extends the package's `BrowsershotFunction` to `BrowsershotLambda`.
But that's quite a lot of work "just" for adding warming to the function.

Or we could introduce a config file, where user can define the number of instance they would like to be warmed. 
In the same step, we could then also add config settings for `runtime`, `memory` and `layers`. As I think this could be stuff users would like to change too.
Adding a config file for such a "simple" package seems almost overkill.

But as Sidecar is currently not that flexible, I think adding a `config/sidecar-browsershot.php` config file is the simplest and most straightforward solution.
If you would like to contribute to the project, feel free to submit a PR adding such a config. If you currently don't have the time, no problem! I would then implement this in the next few days.

Created an issue for this: https://github.com/stefanzweifel/sidecar-browsershot/issues/17

**Reply by aarondfrancis** [reply]
Let me know if I need to change anything on the Sidecar end to accommodate this!

**Reply by stefanzweifel** [reply]
I've added a new `sidebar-browsershot` config in #18 which also contains a `warming` config.
https://github.com/stefanzweifel/sidecar-browsershot#warming

@Jamesking56 a [new release](https://github.com/stefanzweifel/sidecar-browsershot/releases/tag/v1.1.0) has been created. So you should be able to just `composer update wnx/sidecar-browsershot` and then set `SIDECAR_BROWSERSHOT_WARMING_INSTANCES=5` in your `.env` file and be good to go.

**Reply by Jamesking56** [reply]
Thanks @stefanzweifel for the speedy update! 👍
